includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -680,7 +680,7 @@ protected function update_version_and_type( &$product ) {
 	 */
 	protected function clear_caches( &$product ) {
 		wc_delete_product_transients( $product->get_id() );
-		wp_cache_delete( 'product-' . $product->get_id(), 'products' );
+		wp_cache_delete( 'object-' . $product->get_id(), 'products' );
```

Comment:
What I'm getting at is this:
```
$product = WC()->product_factory->get_product( $product_id );
$product->set_width( 10 );
$product->save();
$updated_product = WC()->product_factory->get_product( $product_id );
```

`$updated_product` will be the unmodified, previously-cached product since it caches with a `product-{ID}` key in the `products` group and `clear_caches` will delete with an `object-{ID}` key in the `products` group.

Line 51 class-wc-product-factory.php: `				wp_cache_set( 'product-' . $product_id, $product, 'products' );
`